### ISSUE TITLE
typeof(object).IsAssignableFrom(typeof(byte*)) should be false

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtimelab#189]
JIT/Intrinsics/TypeIntrinsics_r

**Comment by MichalStrehovsky** [comment]
The best way to approach this is to study the developer workflow document https://github.com/dotnet/runtimelab/blob/feature/NativeAOT/docs/workflow/building/coreclr/nativeaot.md and set up the repro project to have this in it:

```csharp
Console.WriteLine(typeof(object).IsAssignableFrom(typeof(byte*)));
```

Then follow the entire Visual Studio workflow until you have reproNative open in Visual Studio - once you have that, open the Program.cs file and set a breakpoint on this line and start debugging. Debug through the entire code until you find the culprit.

The code should print "false", but it prints "true" right now.

**Comment by ThomasGoulet73** [comment]
Hi @MichalStrehovsky, I thought this issue was a good first issue in this repo so I investigated and I was able to reproduce/debug the bug. I may have found the culptrit [here](https://github.com/dotnet/runtimelab/blob/feature/NativeAOT/src/coreclr/src/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/Assignability.cs#L105). I also noticed that the code was copied/vastly inspired by the code [here](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/Assignability.cs#L104), which has the same bug.

Both methods returns true when the type From is a pointer and the type To is object:
``````
if (fromTypeInfo.IsPointer)
{
	if (toTypeInfo.Equals(CommonRuntimeTypes.Object))
		return true;  // T* is castable to Object.

	...
}
``````

The following code shows the same error when using the MetadataLoadContext:
``````
PathAssemblyResolver resolver = new PathAssemblyResolver(new string[] { typeof(object).Assembly.Location });
using (MetadataLoadContext context = new MetadataLoadContext(resolver, "System.Private.CoreLib"))
{
	Assembly assembly = context.LoadFromAssemblyName("System.Private.CoreLib");

	Type fromType = assembly.GetType("System.Byte*");
	Type toType = assembly.GetType("System.Object");

	bool result = toType.IsAssignableFrom(fromType) == typeof(object).IsAssignableFrom(typeof(byte*)); //Returns false, should return true
}
``````

The same bug also applies to `typeof(UIntPtr).IsAssignableFrom(typeof(byte*))`, which returns false without NativeAOT and true with NativeAOT.

If it really is the culprit of the bug, I can create a PR in this repo to remove the check and then file a separate issue in the runtime repo for the bug in MetadataLoadContext.

Thanks
